File: nodejs/src/decorator/MilkDecorator.ts
import BeverageDecorator from "./BeverageDecorator";

class MilkDecorator extends BeverageDecorator {
    getDescription(): string {
        return `${this.beverage.getDescription()} + Sữa`;
    }
    getCost(): number {
        return this.beverage.getCost() + 10000;
    }
}

export default MilkDecorator;

File: spring/demo/src/main/java/com/example/demo/decorator/MilkDecorator.java
package com.example.demo.decorator;

public class MilkDecorator extends BeverageDecorator {
    public MilkDecorator(Beverage beverage) {
        super(beverage);
    }

    @Override
    public String getDescription() {
        return beverage.getDescription() + " + Sữa";
    }

    @Override
    public int getCost() {
        return beverage.getCost() + 10000;
    }
}